config/include.go

```diff
@@ -264,6 +264,8 @@ func (targetConfig *TerragruntConfig) Merge(sourceConfig *TerragruntConfig, terr
 	if sourceConfig.Inputs != nil {
 		targetConfig.Inputs = mergeInputs(sourceConfig.Inputs, targetConfig.Inputs)
 	}
+
+	targetConfig.ProcessedIncludes = sourceConfig.ProcessedIncludes
```

Comment:
I think this can be removed. By moving the setting of the `ProcessedIncludes` to where it is now, it should be set after all the merge operations have been handled. Same with the equivalent setting for `DeepMerge`.